File: solutions/scala/problem-00020.scala
/*
[minhuang@mocha:scala]$ time scala problem-00020.scala 
648

real  0m1.227s
user  0m0.248s
sys 0m0.064s
*/
import scala.Math._

def fact(n: BigInt): BigInt = {
  def recur(n: BigInt, acc: BigInt): BigInt = if (n == 1) acc else recur(n - 1, n * acc)
  if (n == 0) 1 else recur(n, 1)
}

val sum = fact(100).toString.map(_.toString.toInt).reduceLeft(_+_)
println(sum)

File: solutions/ruby/problem-00020.rb
#!/usr/bin/env ruby
# [minhuang@mocha:ruby]$ time ruby problem-00020.rb 
# 648
#
# real  0m0.022s
# user  0m0.012s
# sys 0m0.004s
puts (1..100).inject(:*).to_s.chars.map { |c| c.to_i }.inject(:+)
